types/validator.go

```diff
@@ -159,7 +159,10 @@ func ValidatorFromProto(vp *cmtproto.Validator) (*Validator, error) {
 
 	pk, err := ce.PubKeyFromTypeAndBytes(vp.PubKeyType, vp.PubKeyBytes)
 	if err != nil {
-		return nil, err
+		pk, err = ce.PubKeyFromProto(vp.PubKey)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
In this case, before continuing I think we should:
* Extract the keytype (string) from the public key itself
* Assign the keytype to `pub_key_type`
* Nullify pub_key

Reply:
When the state is updated, it will be saved in a new format. The error is only returned upon loading the state from the state store (i.e., when we're loading the old state using Comet v1).